### PR TITLE
MGDAPI-1032 Migrate operator-sdk to 1.2.0

### PR DESCRIPTION
**Pull request by sergioifg94**
# Description

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: controllers/namespacelabel/namespacelabel_controller.go [hunk 1]
```diff
@@ -69,22 +63,22 @@ type network struct {
...
-// Add creates a new namespacelabel Controller and adds it to the Manager. The Manager will set fields on the Controller
-// and Start it when the Manager is Started.
-func Add(mgr manager.Manager) error {
-	reconcile, err := newReconciler(mgr)
-	if err != nil {
-		return err
-	}
-	return add(mgr, reconcile)
-}
+// // Add creates a new namespacelabel Controller and adds it to the Manager. The Manager will set fields on the Controller
```
Comment (philbrookes) [inline controllers/namespacelabel/namespacelabel_controller.go:66]: Let's just delete this if it's not needed any longer

File: controllers/namespacelabel/namespacelabel_controller.go [hunk 2]
```diff
@@ -97,112 +91,123 @@ var (
...
-	if err != nil {
-		return err
-	}
-
-	return nil
 }
 
-// blank assignment to verify that ReconcileNamespace implements reconcile.Reconciler
-var _ reconcile.Reconciler = &ReconcileNamespaceLabel{}
+// // newReconciler returns a new reconcile.Reconciler
```
Comment (philbrookes) [inline controllers/namespacelabel/namespacelabel_controller.go:126]: Same here, as a general rule - let's delete code instead of commenting it out.

File: controllers/namespacelabel/namespacelabel_controller.go [hunk 3]
```diff
@@ -97,112 +91,123 @@ var (
...
 	namespaceSegments := strings.Split(watchNS, "-")
 	namespacePrefix := strings.Join(namespaceSegments[0:2], "-") + "-"
 	operatorNs := namespacePrefix + "operator"
 
-	return &ReconcileNamespaceLabel{
-		mgr:               mgr,
-		client:            mgr.GetClient(),
-		scheme:            mgr.GetScheme(),
+	return &NamespaceLabelReconciler{
+		Client:            mgr.GetClient(),
```
Comment (philbrookes) [inline controllers/namespacelabel/namespacelabel_controller.go:119]: Why did these need to be converted to public attributes?

File: controllers/namespacelabel/namespacelabel_controller.go [hunk 4]
```diff
@@ -278,7 +283,7 @@ func Uninstall(v string, request reconcile.Request, r *ReconcileNamespaceLabel)
 
 	if rhmiCr.DeletionTimestamp == nil {
 		r.log.Info("Deleting RHMI CR")
-		err := r.client.Delete(r.context, rhmiCr)
+		err := r.Delete(context.TODO(), rhmiCr)
```
Comment (philbrookes) [inline controllers/namespacelabel/namespacelabel_controller.go:286]: Why have we converted back to using context.TODO instead of the passed around context?

File: controllers/rhmi/rhmi_controller.go [hunk 5]
```diff
@@ -888,21 +829,161 @@ func (r *ReconcileInstallation) handleCROConfigDeletion(rhmi integreatlyv1alpha1
...
 		return fmt.Errorf("error occurred trying to delete cro config map, %w", err)
 	}
 
 	return nil
 }
 
-func (r *ReconcileInstallation) addCustomInformer(crd runtime.Object, namespace string) error {
+func (r *RHMIReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Creates a new managed install CR if it is not available
+	// kubeConfig := controllerruntime.GetConfigOrDie()
```
Comment (philbrookes) [inline controllers/rhmi/rhmi_controller.go:848]: delete this

File: controllers/rhmiconfig/rhmiconfig_controller.go [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+type RHMIConfigReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=integreatly.org,resources=rhmiconfigs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=integreatly.org,resources=rhmiconfigs/status,verbs=get;update;patch
+
+func (r *RHMIConfigReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
```
Comment (philbrookes) [inline controllers/rhmiconfig/rhmiconfig_controller.go:49]: What is the purpose of this?

File: controllers/namespacelabel/namespacelabel_controller.go (hunk 3 above)
Comment (sergioifg94) [reply controllers/namespacelabel/namespacelabel_controller.go:119]: In the case of `Client`, controllers generated by operator-sdk have now the client as an embedded field, so I folowed that pattern. I'll double check why I converted `Scheme`

File: controllers/subscription/subscription_controller.go [hunk 7]
```diff
@@ -51,20 +49,18 @@ var subscriptionsToReconcile []string = []string{
...
-	reconciler, err := newReconciler(mgr)
-	if err != nil {
-		return err
-	}
-	return add(mgr, reconciler)
-}
-
-func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
-	watchNS, err := k8sutil.GetWatchNamespace()
+// func Add(mgr manager.Manager) error {
```
Comment (philbrookes) [inline controllers/subscription/subscription_controller.go:54]: delete this.

File: controllers/namespacelabel/namespacelabel_controller.go (hunk 4 above)
Comment (sergioifg94) [reply controllers/namespacelabel/namespacelabel_controller.go:286]: If I remember correctly, we stopped storing the context in structs as part of the work to fix the memory leak a while ago. I think the change was originally missed here

File: controllers/rhmiconfig/rhmiconfig_controller.go (hunk 6 above)
Comment (sergioifg94) [reply controllers/rhmiconfig/rhmiconfig_controller.go:49]: It comes in the Reconcile when scaffolding a new controller. I kept it in case we'd investigate at some point using that context instead of `context.TODO()` as we're currently doing, but it's not being used at this point

File: Makefile [hunk 8]
```diff
@@ -254,12 +264,24 @@ test/rhoam/products:
 	mkdir -p $(TEST_RESULTS_DIR)
 	delorean pipeline product-tests --test-config ./test-containers-managed-api.yaml --output $(TEST_RESULTS_DIR) --namespace test-products
 
+
+.PHONY: cluster/deploy
+cluster/deploy: kustomize cluster/cleanup cluster/cleanup/crds cluster/prepare/crd cluster/prepare deploy/integreatly-rhmi-cr.yml
+	@ - oc create -f config/rbac/service_account.yaml
+	@ - cd config/manager && $(KUSTOMIZE) edit set image controller=${IMAGE_FORMAT}
+	@ - $(KUSTOMIZE) build config/redhat-$(INSTALLATION_SHORTHAND) | kubectl apply -f -
```
Comment (jjaferson) [inline Makefile:272]: Replace `kubectl` to `oc`, tests are failing in PROW due to that.

```/go/bin/kustomize build config/crd | kubectl apply -f -
/bin/bash: kubectl: command not found
make: *** [cluster/prepare/crd] Error 127
{"component":"entrypoint","error":"wrapped process failed: exit status 2","file":"prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2021-02-10T15:02:43Z"}
error: failed to execute wrapped command: exit status 2
2021/02/10 15:02:45 Container test in pod rhoam-e2e-test failed, exit code 1, reason Error```

**Review by jjaferson — CHANGES_REQUESTED** [review]

File: Makefile [hunk 9]
```diff
@@ -386,34 +396,34 @@ cluster/cleanup/crds:
 	@-oc delete crd grafanas.integreatly.org
 	@-oc delete crd rhmis.integreatly.org
 	@-oc delete crd webapps.integreatly.org
-	@-oc delete crd rhmiconfigs.integreatly.org
+	@-oc delete crd rhmiconfigs.integreatly.or
```
Comment (Boomatang) [inline Makefile:399]: This seems odd. Should it be `.org` and not `.or`?

File: Makefile [hunk 10]
```diff
@@ -9,11 +9,10 @@ PKG=github.com/integr8ly/integreatly-operator
 TEST_DIRS?=$(shell sh -c "find $(TOP_SRC_DIRS) -name \\*_test.go -exec dirname {} \\; | sort | uniq")
 TEST_POD_NAME=integreatly-operator-test
 COMPILE_TARGET=./tmp/_output/bin/$(PROJECT)
-OPERATOR_SDK_VERSION=0.17.1
+OPERATOR_SDK_VERSION=1.2.0
 AUTH_TOKEN=$(shell curl -sH "Content-Type: application/json" -XPOST https://quay.io/cnr/api/v1/users/login -d '{"user": {"username": "$(QUAY_USERNAME)", "password": "$(QUAY_PASSWORD)"}}' | jq -r '.token')
 TEMPLATE_PATH="$(shell pwd)/templates/monitoring"
-IN_PROW="false"
-TYPE_OF_MANIFEST ?= master
+IN_PROW="false
```
Comment (sergioifg94) [inline Makefile:15]: Typo

File: Makefile [hunk 11]
```diff
@@ -93,32 +91,48 @@ define wait_command
...
-	@-oc create -f deploy/service_account.yaml -n $(NAMESPACE)
-	@cat deploy/$(INSTALLATION_PREFIX)/role_binding.yaml | sed "s/namespace: integreatly/namespace: $(NAMESPACE)/g" | oc replace --force -f -
+	@-oc create -f config/rbac/service_account.yaml -n $(NAMESPACE)
+	@$(KUSTOMIZE) build config/rbac-$(INSTALLATION_SHORTHAND) | oc replace --force -f -	
 	@oc login --token=$(shell oc serviceaccounts get-token rhmi-operator -n ${NAMESPACE}) --server=$(shell sh -c "oc cluster-info | grep -Eo 'https?://[-a-zA-Z0-9\.:]*'") --kubeconfig=TMP_SA_KUBECONFIG --insecure-skip-tls-verify=true
 
 .PHONY: setup/git/hooks
 setup/git/hooks:
-	git config core.hooksPath .githooks
+	git config core.hooksPath .githook
```
Comment (sergioifg94) [inline Makefile:127]: Typo

File: Makefile [hunk 12]
```diff
@@ -433,8 +443,71 @@ vendor/check: vendor/fix
 .PHONY: vendor/fix
 vendor/fix:
 	go mod tidy
-	go mod vendor
+	go mod vendo
```
Comment (sergioifg94) [inline Makefile:446]: Typo

File: Makefile (hunk 9 above)
Comment (Boomatang) [inline Makefile:399]: typo?

File: Makefile [hunk 13]
```diff
@@ -433,8 +443,71 @@ vendor/check: vendor/fix
...
+
+.PHONY: manifest/release
```
Comment (MStokluska) [inline Makefile:448]: hey guys, I think this needs a rebase, the manifest/release was replaced with manifest/prodsec, without it both the prepare-release pipeline and the manifest check on this PR will fail.

File: controllers/namespacelabel/namespacelabel_controller.go (hunk 4 above)
Comment (philbrookes) [reply controllers/namespacelabel/namespacelabel_controller.go:237]: Yeah, we shouldn't be storing the context, but it should still be passed around from the rhmi controller, just not stored anywhere. Creating new ones everywhere means that we lose the ability to abort a reconcile if required.

Comment (philbrookes) [reply controllers/namespacelabel/namespacelabel_controller.go:237]: e.g. at the beginning of the uninstall reconcile, we abort any - potential - running install reconcilers. This was to address a bug where the uninstall would go faster than a running install reconcile, and the still running install reconcile would put everything back after it got deleted.

File: pkg/products/threescale/clients_test.go [hunk 14]
```diff
@@ -52,17 +50,17 @@ func getSigClient(preReqObjects []runtime.Object, scheme *runtime.Scheme) *clien
...
+			// 	cs := &coreosv1alpha1.CatalogSource{
+			// 		ObjectMeta: metav1.ObjectMeta{
+			// 			Name:      "test-catalogsource",
+			// 			Namespace: obj.GetTargetNamespace(),
+			// 		},
+			// 	}
+			// 	err := sigsFakeClient.GetSigsClient().Create(ctx, cs)
+			// 	if err != nil {
+			// 		return err
+			// 	}
```
Comment (KevFan) [inline pkg/products/threescale/clients_test.go:63]: Commeneted out case still needed ?

File: controllers/namespacelabel/namespacelabel_controller.go [hunk 15]
```diff
@@ -97,112 +81,84 @@ var (
...
 }
 
+// +kubebuilder:rbac:groups="",resources=namespaces,verbs=list;get;watch;update
+
 // Reconcile : The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
-func (r *ReconcileNamespaceLabel) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+func (r *NamespaceLabelReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+	// log := r.Log.WithValues("namespacelabel", request.NamespacedName)
```
Comment (KevFan) [inline controllers/namespacelabel/namespacelabel_controller.go:134]: Log not needed anymore ? :thinking:

File: test/scripts/backup-restore/j07-verify-rhsso-backup-and-restore.sh [hunk 16]
```diff
@@ -1,17 +0,0 @@
...
-export AWS_SECRET_ACCESS_KEY=$(oc get secret aws-creds -n kube-system -o jsonpath={.data.aws_secret_access_key} | base64 --decode)
-export AWS_ACCESS_KEY_ID=$(oc get secret aws-creds -n kube-system -o jsonpath={.data.aws_access_key_id} | base64 --decode)
-AWS_DB_ID=$(oc get secret/keycloak-db-secret -o go-template --template="{{.data.POSTGRES_EXTERNAL_ADDRESS|base64decode}}" -n ${NS_PREFIX}-rhsso | awk -F\. '{print $1}')
-AWS_REGION=$(oc get infrastructure cluster -o jsonpath='{.status.platformStatus.aws.region}')
-RHMI_CR_NAME=$(oc get rhmi -n ${NS_PREFIX}-operator -o json | jq -r '.items[0].metadata.name')
-POSTGRES_CR_NAME="rhsso-postgres-$RHMI_CR_NAME"
-DATABASE_SECRET="rhsso-postgres-$RHMI_CR_NAME"
-
-# Perform the test
-test_postgres_backup $POSTGRES_CR_NAME $DATABASE_SECRET $AWS_DB_ID $AWS_REGION $NS_PREFIX
```
Comment (KevFan) [inline test/scripts/backup-restore/j07-verify-rhsso-backup-and-restore.sh:17]: Best to keep this as it's used in the relevant test case
* https://github.com/integr8ly/integreatly-operator/blob/master/test-cases/tests/backup-restore/j07b-verify-cluster-sso-backup-and-restore.md

File: test/scripts/backup-restore/j05-verify-3scale-postgres-backup-and-restore.sh [hunk 17]
```diff
@@ -1,28 +0,0 @@
...
-export AWS_SECRET_ACCESS_KEY=$(oc get secret aws-creds -n kube-system -o jsonpath='{.data.aws_secret_access_key}' | base64 --decode)
-export AWS_ACCESS_KEY_ID=$(oc get secret aws-creds -n kube-system -o jsonpath='{.data.aws_access_key_id}' | base64 --decode)
-AWS_DB_ID=$(oc get secret/system-database -o go-template --template="{{.data.URL|base64decode}}" -n ${NS_PREFIX}-3scale | $grep_cmd -Po "(?<=@).*?(?=\.)")
-AWS_REGION=$(oc get infrastructure cluster -o jsonpath='{.status.platformStatus.aws.region}')
-RHMI_CR_NAME=$(oc get rhmi -n ${NS_PREFIX}-operator -o json | jq -r '.items[0].metadata.name')
-POSTGRES_CR_NAME="threescale-postgres-$RHMI_CR_NAME"
-DATABASE_SECRET="threescale-postgres-$RHMI_CR_NAME"
-
-# Perform the test
-test_postgres_backup $POSTGRES_CR_NAME $DATABASE_SECRET $AWS_DB_ID $AWS_REGION $NS_PREFIX
```
Comment (KevFan) [inline test/scripts/backup-restore/j05-verify-3scale-postgres-backup-and-restore.sh:28]: Probably best to keep this as it's used in revelant test case
* https://github.com/integr8ly/integreatly-operator/blob/master/test-cases/tests/backup-restore/j05b-verify-3scale-backup-and-restore.md

File: test/scripts/backup-restore/j08-verify-user-sso-backup-and-restore.sh [hunk 18]
```diff
@@ -1,17 +0,0 @@
...
-export AWS_SECRET_ACCESS_KEY=$(oc get secret aws-creds -n kube-system -o jsonpath={.data.aws_secret_access_key} | base64 --decode)
-export AWS_ACCESS_KEY_ID=$(oc get secret aws-creds -n kube-system -o jsonpath={.data.aws_access_key_id} | base64 --decode)
-AWS_DB_ID=$(oc get secret/keycloak-db-secret -o go-template --template="{{.data.POSTGRES_EXTERNAL_ADDRESS|base64decode}}" -n ${NS_PREFIX}-user-sso | awk -F\. '{print $1}')
-AWS_REGION=$(oc get infrastructure cluster -o jsonpath='{.status.platformStatus.aws.region}')
-RHMI_CR_NAME=$(oc get rhmi -n ${NS_PREFIX}-operator -o json | jq -r '.items[0].metadata.name')
-POSTGRES_CR_NAME="rhssouser-postgres-$RHMI_CR_NAME"
-DATABASE_SECRET="rhssouser-postgres-$RHMI_CR_NAME"
-
-# Perform the test
-test_postgres_backup $POSTGRES_CR_NAME $DATABASE_SECRET $AWS_DB_ID $AWS_REGION $NS_PREFIX
```
Comment (KevFan) [inline test/scripts/backup-restore/j08-verify-user-sso-backup-and-restore.sh:17]: I think best to keep this as it's used in backup & restore test case
*  https://github.com/integr8ly/integreatly-operator/blob/master/test-cases/tests/backup-restore/j08b-verify-user-sso-backup-and-restore.md

File: test/scripts/backup-restore/postgres.sh [hunk 19]
```diff
@@ -1,235 +0,0 @@
...
-  done
-
-  # Create the dump
-  kubectl exec deploy/throw-away-postgres \
-    -n ${NS_PREFIX}-operator \
-    -- env PGPASSWORD=$DB_PASSWORD pg_dump -h $DB_HOST -p $DB_PORT -U $DB_USER $DATABASE_NAME > $DUMP_FILE
-
-  # Delete the throwaway postgres
-  oc delete postgres/throw-away-postgres -n ${NS_PREFIX}-operator
-}
```
Comment (KevFan) [inline test/scripts/backup-restore/postgres.sh:235]: This script is used as the subscript by the following backup & restore test cases, so proabably best to keep this :thinking: 
* https://github.com/integr8ly/integreatly-operator/tree/master/test-cases/tests/backup-restore

File: test/e2e/suite_test.go [hunk 20]
```diff
@@ -0,0 +1,232 @@
...
+			Expect(err).NotTo(HaveOccurred())
+		}
+	}
+
+	err = testEnv.Stop()
+	Expect(err).ToNot(HaveOccurred())
+})
+
+func waitForInstallationStageCompletion(k8sClient client.Client, retryInterval, timeout time.Duration, phase string) error {
+	err := wait.Poll(time.Second*5, time.Minute*30, func() (done bool, err error) {
```
Comment (KevFan) [inline test/e2e/suite_test.go:183]: Doesn't look to use the `retryInterval` and `timeout time.Duration` parameters in the poll, so maybe either set these or remove as part of function if we're looking to have this the same for all stages :thinking:

File: test/README.md [hunk 21]
```diff
@@ -1,77 +0,0 @@
...
-
-```
-make image/functional/build
-```
-
-and push it to quay.io:
-
-```
-make image/functional/push
-```
```
Comment (KevFan) [inline test/README.md:77]: Just wondering is this delete of test docs / OWNERS intended and we're sure about this ? Think there was an effort to [document best practices for functional tests](https://github.com/integr8ly/integreatly-operator/pull/504 ) and removing the owners here will remove the QE guys for approving changes in the `/tests` directory  :thinking:

**Review by KevFan — COMMENTED** [review]
Looks good :+1:  Just have some small comments / questions

File: controllers/namespacelabel/namespacelabel_controller.go (hunk 15 above)
Comment (sergioifg94) [reply controllers/namespacelabel/namespacelabel_controller.go:134]: That should've been removed. It's the default logger created when scaffolding with operator-sdk, but we removed that in favour of the internal logger implementations in the `resources` package. I'll remove the line

File: test/scripts/backup-restore/j07-verify-rhsso-backup-and-restore.sh (hunk 16 above)
Comment (sergioifg94) [reply test/scripts/backup-restore/j07-verify-rhsso-backup-and-restore.sh:17]: I must've missed the whole `backup-restore` folder when moving the directories to the new project. Good catch!

File: controllers/namespacelabel/namespacelabel_controller.go [hunk 22]
```diff
@@ -0,0 +1,317 @@
...
+
+// +kubebuilder:rbac:groups="",resources=namespaces,verbs=list;get;watch;update
+
+// Reconcile : The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *NamespaceLabelReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+	// log := r.Log.WithValues("namespacelabel", request.NamespacedName)
+
+	if request.NamespacedName.Name == r.operatorNamespace {
```
Comment (KevFan) [inline controllers/namespacelabel/namespacelabel_controller.go:136]:
```suggestion
	if request.Namespace == r.operatorNamespace {
```
Probably due to the operator-sdk update, but `request.NamespacedName.Name` doesn't look to be namespace name anymore but is the name of the watched config map name (can be checked by debugging with a breakpoint at this line). 

Without the above change (or to `request.NamespacedName.Namespace`), this block of code won't be run anymore so probably would affect addon deletion uninstall flow :thinking:

Comment (sergioifg94) [reply controllers/namespacelabel/namespacelabel_controller.go:136]: Interesting stuff. I think that this is unrelated to the operator-sdk but a bug that was introduced when adding the ConfigMap watch to this controller. The `ctrl.Request` struct embeds `types.NamespacedName` which means that the `.NamespacedName` is actually redundant:
```go
type Request struct {
	// NamespacedName is the name and namespace of the object to reconcile.
	types.NamespacedName
}
```

`request.Namespace` would work for the ConfigMap events but I'm not sure if it'd work with Namespace resources as they're cluster scoped. I think ideally we'd have 2 separate controllers, one for the namespace that would use `request.Name` and one for the ConfigMap that could watch specifically the ConfigMap we're looking for, and the common logic could be factored in a separate package.

A quicker fix would be to simply check for both, but it sounds pretty ugly to me:
```go
if request.Name == r.operatorNamespace || request.Namespace == r.operatorNamespace {
```

Comment (sergioifg94) [reply controllers/namespacelabel/namespacelabel_controller.go:136]: @KevFan If it sounds good to you, I'll keep it as it is in this PR and create a separate ticket to improve the controller. I think the reason it still works is because the controller is reconciling every minute, so changes in the ConfigMap will be spotted eventually

Comment (KevFan) [reply controllers/namespacelabel/namespacelabel_controller.go:136]: @sergioifg94 Sure, sounds good to me once it's flagged :+1:

File: controllers/namespacelabel/namespacelabel_controller.go (hunk 4 above)
Comment (philbrookes) [reply controllers/namespacelabel/namespacelabel_controller.go:236]: I've just noticed that this is in the namespace label controller. So this can be ignored. So long as all the product reconcilers are still being passed a context from the installation controller.

Comment (philbrookes) [reply controllers/namespacelabel/namespacelabel_controller.go:236]: actually, this information appears to be an out of date, so disregard this thread.

**Comment by philbrookes** [comment]
:fireworks: :champagne:

**Comment by jjaferson** [comment]
/test rhoam-e2e

timed out on `C03 - Verify that alerting mechanism works` waiting for the condition

**Comment by sergioifg94** [comment]
> **A18 - Verify RHMI Config CRs Successful**
> Expected mutating webhook to add "system:admin" lastEditUsername annotation to RHMIConfig. Got system:serviceaccount:redhat-rhoam-operator:rhmi-operator instead

This test makes a change on the RHMIConfig and verifies that the webhook sets the `lastEditUsername` annotation. Seems like the annotation references the operator SA so my guess is that the this is an edge case where the operator happened to reconcile the RHMIConfig after the CR was updated by the test so the annotation was overwritten to the operator SA.

/test rhoam-e2e
